docs/SMT.md

```diff
@@ -378,10 +378,21 @@ This backstepping process allows the traversal to continue until it reaches a
 sentinel leaf that has the longest common prefix and most bits in common with
 the provided hash, up to the depth of the leaf found.
 
-This method guarentees a proof of inclusion in all cases and can be verified by
+This method guarantees a proof of inclusion in all cases and can be verified by
 using the `VerifyClosestProof` function which requires the proof and root hash
 of the trie.
 
+NB: If the hash provided to the `ClosestProof` function is known prior to the
+tree being filled and closed there is the possibility of placing a leaf where
+the hash will lead. If used **as intended** the hash provided should **not** be
+known prior to calling the method and the tree should not be updateable after
+the fact. If the hash were known prior to closing the tree and not inserting
+into it anymore a leaf could be inserted into the tree in such a way that it
+would always be produced from the `ClosestProof` method (as it is deterministic).
```

Comment:
Not sure if this is correct but I'm still struggling with this part. Can you rephrase "procuded from", I'm not sure what that means in this context.

```suggestion
the fact. If the last hash were known by <whom?>, after the last update but prior to closing, a leaf could be inserted into the tree in such a way that it would always be produced from the `ClosestProof` method (as it is deterministic).
```

Reply:
So essentially the caller of the method could place a leaf that would be guareneteed to be returned by the method IFF they know they hash prior to losing access to inserting into the tree. Have tried to make this clearer while keeping it general as this *could* be used in different ways from how we use it